kernel/src/common/regs/README.md

```diff
@@ -0,0 +1,207 @@
+# Register/Bitfield Interface for Tock
+
+This module provides an interface for defining and manipulating memory mapped
+registers and bitfields.
+
+## Defining registers
+
+The module provides three types for working with memory mapped registers: `ReadWrite`,
+`ReadOnly`, and `WriteOnly`, providing read-write, read-only, and write-only functionality,
+respectively.
+
+Defining the registers is similar to the C-style approach, where each register
+is a field in a packed struct:
+
+```rust
+use common::regs::{ReadOnly, ReadWrite, WriteOnly};
+
+#[repr(C, packed)]
+struct Registers {
+    // Control register: read-write
+    // The 'Control' parameter constrains this register to only use fields from
+    // a certain group (defined below in the bitfields section).
+    cr: ReadWrite<u8, Control>,
+
+    // Status register: read-only
+    s: ReadOnly<u8, Status>
+
+    // Registers can be bytes, halfwords, or words:
+    // Note that the second type parameter can be omitted, meaning that there
+    // are no bitfields defined for these registers.
+    byte0: ReadWrite<u8>,
+    byte1: ReadWrite<u8>,
+    short: ReadWrite<u16>,
+    word: ReadWrite<u32>
+
+    // Etc.
+}
+```
+
+## Defining bitfields
+
+Bitfields are defined through the `bitfields!` macro:
+
+```rust
+bitfields! [
+    // First parameter is the register width for the bitfields. Can be u8, u16,
+    // or u32.
+    u8,
```

Comment:
Given the example above, the registers have different sizes. Does it make more sense for this to be inside of each register description below? This is closely related to whether you can call `bitfields!` multiple times.

Reply:
The reason I put it as the first parameter is to minimize the amount of redundant typing needed for someone to define the bitfields. Since the macro can be called multiple times in the same file, the intention was that you define all same-width registers together.